src/api.rs

```diff
@@ -1244,16 +1244,17 @@ impl<T: Pixel> ContextInner<T> {
 
           update_rec_buffer(fi, fs);
 
-          // Copy the `rec_buffer` into subsequent frame invariants.
+          // Copy persistent fields into subsequent frame invariants.
```

Comment:
Are these fields actually used by anything in the lookahead? Technically for everything except the next frame to be encoded, they are set to incorrect values.

Reply:
I think in the lookahead I will actually be setting them to different values (original frames) to find MVs.

Reply:
Yeah OK.

Also technically I think you only need to set one fi in the future for correct encoding?

Reply:
Yeah, I thought about this. Let me try it.

Reply:
Well, turns out this breaks `test_encode_decode::reordering_aom`.

Reply:
Probably because we aren't writing into the correct fi ahead.

This isn't computationally expensive, so I guess maybe it's okay to leave it until these get split out by lookahead.

Reply:
Though they are in order by output_frameno, so it's kinda weird.

Reply:
What I did was taking this `for subsequent_fi in` and making it into an `if let Some(subsequent_fi) = chain.next() { ... }`.

Reply:
OK I *think* that's correct (it should receive frame_invariants[output_frameno + 1]
Might have to do with not getting past invalid fis or something?